manifests/exporter.pp

```diff
@@ -63,7 +71,26 @@
     nfstag      => 'rsan',
   }
 
-
+} else {
+    nfs::server::export{ '/var/log/':
+    ensure      => 'absent',
+    mount       => "/var/pesupport/${facts['fqdn']}/log",
+    options_nfs => 'tcp,nolock,rsize=32768,wsize=32768,soft,noatime,actimeo=3,retrans=1',
+    nfstag      => 'rsan',
+  }
+  nfs::server::export{ '/opt/puppetlabs/':
+    ensure      => 'absent',
+    mount       => "/var/pesupport/${facts['fqdn']}/opt",
+    options_nfs => 'tcp,nolock,rsize=32768,wsize=32768,soft,noatime,actimeo=3,retrans=1',
+    nfstag      => 'rsan',
+  }
+  nfs::server::export{ '/etc/puppetlabs/':
+    ensure      => 'absent',
+    mount       => "/var/pesupport/${facts['fqdn']}/etc",
+    options_nfs => 'tcp,nolock,rsize=32768,wsize=32768,soft,noatime,actimeo=3,retrans=1',
+    nfstag      => 'rsan',
+  }
+}
```

Comment:
We could save a fair amount of code here by using the conditional to set a variable, then use variables in the resources, rather than duplicating the resources themselves e.g.

```puppet
  $ensure = $nfsmount ? {
    true  => 'mounted',
    false => 'absent,
  }

  nfs::server::export{ '/etc/puppetlabs/':
    ensure      => $ensure,
    clients     => $clients,
    mount       => "/var/pesupport/${facts['fqdn']}/etc",
    options_nfs => 'tcp,nolock,rsize=32768,wsize=32768,soft,noatime,actimeo=3,retrans=1',
    nfstag      => 'rsan',
  }
```

Reply:
sage counsel as always, ill make it so